flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/factories/TableFactoryUtil.java

```diff
@@ -103,4 +103,15 @@
 		return Optional.empty();
 	}
 
+	/**
```

Comment:
We should make sure `TableSinkFactory#createTableSink(context)` is the only entry to be invoked by the planner. However,  the above `createTableSinkForCatalogTable` method still calling the old `createTableSink` method in old planner.

Reply:
I'll push another commit to `Support create table source/sink by context in legacy planner`